File: song.py
from db import Db


class Song:
    def __init__(self, object_id, name=None, interpreter=None, album=None, genre=None, release=None):
        if object_id is None:
            self.name = name
            self.interpreter = interpreter
            self.album = album
            self.genre = genre
            self.release = release
            db = Db().access()
            db['songs'].insert_one(
                {"name": name,
                 "interpreter": interpreter,
                 "album": album,
                 "genre": genre,
                 "release": release
                 }
            )
        else:
            self.object_id = object_id
            db = Db().access()
            data = db['songs'].find_one({'_id': object_id})
            self.name = data['name']
            self.interpreter = data['interpreter']
            self.album = data['album']
            self.genre = data['genre']
            self.release = data['release']

    def get_name(self):
        return self.name

    def get_interpreter(self):
        return self.interpreter

    def get_album(self):
        return self.album

    def get_genre(self):
        return self.genre

    def get_release(self):
        return self.release

File: m165-jukebox-webservice/src/main/java/tech/bison/trainee/server/business/domain/song/Song.java
package tech.bison.trainee.server.business.domain.song;

import java.time.ZonedDateTime;
import java.util.Optional;
import java.util.UUID;

public class Song {
  private final UUID id;
  private String name;
  private String interpreter;
  private String album;
  private String genre;
  private ZonedDateTime release;

  private Song(UUID id, String name, String interpreter, String album, String genre, ZonedDateTime release) {
    this.id = id;
    this.name = name;
    this.interpreter = interpreter;
    this.album = album;
    this.genre = genre;
    this.release = release;
  }

  public static Song of(Optional<UUID> id, String name, String interpreter, String album, String genre,
                        ZonedDateTime release) {
    return new Song(id.orElse(UUID.randomUUID()), name, interpreter, album, genre, release);
  }

  public UUID getId() {
    return id;
  }

  public String getName() {
    return name;
  }

  public String getInterpreter() {
    return interpreter;
  }

  public String getAlbum() {
    return album;
  }

  public String getGenre() {
    return genre;
  }

  public ZonedDateTime getRelease() {
    return release;
  }

  public void setName(String name) {
    this.name = name;
  }

  public void setInterpreter(String interpreter) {
    this.interpreter = interpreter;
  }

  public void setAlbum(String album) {
    this.album = album;
  }

  public void setGenre(String genre) {
    this.genre = genre;
  }

  public void setRelease(ZonedDateTime release) {
    this.release = release;
  }
}
